src/octoprint/plugins/action_command_notification/templates/action_command_notification_settings.jinja2

```diff
@@ -16,4 +16,11 @@
             </label>
         </div>
     </div>
+    <div class="control-group" data-bind="css: { error: !regexValid() }">
+        <label class="control-label" for="settings-filterRegex"> {{ _('Filtering Regex') }}</label>
+        <div class="controls">
+            <textarea rows="4" class="block" id="settings-filterRegex" data-bind="value: regex, valueUpdate: 'afterkeydown'"></textarea>
```

Comment:
I think changing the value data-bind here to `settings.plugins.action_command_notification.regex` instead of using the one created in onBeforeBinding will fix the saving issue. And also in that perspective, replacing that in the js side as well would remove the need for onBeforeBinding altogether?

Reply:
But why is the current code not working? I have seen other code doing the same thing. Although, I made this change you suggested but still it doesn't save the settings.

Reply:
Actually, I named the variable incorrectly. I was now using custom binding and used the settingsViewModel as the dependency in the custom ViewModel and stored that passes parameter in the self.settings variable so I must use : `settings.settings.plugins.action_command_notification.regex` and NOT `settings.plugins.action_command_notification.regex`